### ISSUE TITLE
Install fusio/adapter-stripe error. It looks like version not compatible. How can I solve it?

### ISSUE THREAD

**Issue by cococonuts** [apioo/fusio#504]
sudo composer require fusio/adapter-stripe:*
Do not run Composer as root/super user! See https://getcomposer.org/root for details
Continue as root/super user [yes]? yes
./composer.json has been updated
Running composer update fusio/adapter-stripe
Loading composer repositories with package information
Updating dependencies
Your requirements could not be resolved to an installable set of packages.

  Problem 1
    - laminas/laminas-code 4.7.0 requires php >=7.4, <8.2 -> your php version (8.2.5) does not satisfy that requirement.
    - doctrine/migrations 2.3.5 requires friendsofphp/proxy-manager-lts ^1.0 -> satisfiable by friendsofphp/proxy-manager-lts[v1.0.13].
    - fusio/impl v6.3.8 requires doctrine/migrations ^2.3 -> satisfiable by doctrine/migrations[2.3.5].
    - friendsofphp/proxy-manager-lts v1.0.13 requires laminas/laminas-code ~3.4.1|^4.0 -> satisfiable by laminas/laminas-code[4.7.0].
    - fusio/impl is locked to version v6.3.8 and an update of this package was not requested.

Installation failed, reverting ./composer.json and ./composer.lock to their original content.

**Comment by chriskapp** [comment]
Hi @cococonuts could you explicit try to use the version [v5.2.0](https://github.com/apioo/fusio-adapter-stripe/releases/tag/v5.2.0) i.e. with `composer require fusio/adapter-stripe ^5.0` we use this version also in our docker image, at our docker image we still use PHP 8.0. In general we are currently also working on a new release where we update many dependencies and we will then also update the docker PHP version.

**Comment by cococonuts** [comment]
I got the same issue with "sudo composer require fusio/adapter-stripe ^5.0". Is it because I installed php 8.2?

I tried to install adapter-strip with gnore-platform-reqs option. It works and I can config stripe with API key in fusio admin. 
"composer require fusio/adapter-stripe:^5.0 --ignore-platform-reqs"

However, when I purchase subscription in developer app, it goes to error 500.
POST | https://myhost.com/index.php/consumer/payment/stripe/checkout

<img width="1409" alt="image" src="https://user-images.githubusercontent.com/10249233/236548230-7475ae75-d573-439d-9089-d1cb9cd3b428.png">

**Comment by cococonuts** [comment]
Do you have dev document for how to debug the router/issue in fusio?

**Comment by chriskapp** [comment]
Hi @cococonuts, could you check the PHP error logs, there should be an error regarding this issue, I would also assume that this is probably a problem regarding PHP 8.2

**Comment by cococonuts** [comment]
Is the error.log in the /var/log/nginx/?
I just start to use PHP. I am not sure about where is PHP error log. Sorry.
I also tried to modify /etc/php/8.2/fpm/php.ini and  /etc/php/8.2/cli/php.ini to make the error_log=/var/log/nginx/error.log
But the it is still empty in error.log.

**Comment by chriskapp** [comment]
@cococonuts ok this looks indeed like you have an older version installed, could you run in your folder the `composer info` command to see which versions are currently installed? You should have version `v5.4.3` installed of the `fusio/engine` package.

**Comment by cococonuts** [comment]
@chriskapp 
I upgrade the fusio-engine to v5.4.3. It works!

But I have new questions:
why I don't have Contract and invoices in "Monetization"
<img width="395" alt="image" src="https://user-images.githubusercontent.com/10249233/236999040-23eab43a-2114-4d89-81d5-2590bbae24f9.png">

I successfully purchase a subscription and one-time plan. I can't find transaction recording in fusio portal.
What else should I config?

**Comment by chriskapp** [comment]
@cococonuts so great that this works, in a later version we have removed our internal billing system and the Contract/Invoice panel, since all this can be handled by the payment provider i.e. stripe which has much more options to customize your billing setup. You need to create a stripe connection and then register a payment webhook at the stripe portal which calls your Fusio instance at `/system/payment/stripe/webhook`, then you also need to set the webhook secret at the system / settings `payment_stripe_secret`. Then you should see all incoming transactions.

**Comment by cococonuts** [comment]
Got your point.
I have set up webhook on stripe website and payment_stripe_secret on fusio admin portal.
But I have not got transactions data.

I guess perhaps I didn't set up endpoint on strip webhook.
My fusio app link is: https://myhost.com/apps/fusio
The actual https router is  https://myhost.com/index.php/

So which endpoint in stripe webhook should be set:
 https://myhost.com/apps/fusio/system/payment/stripe/webhook
or 
https://myhost.com/index.php/system/payment/stripe/webhook

**Comment by chriskapp** [comment]
yes, so you would need to use the url which points directly to your Fusio instance, in your case:
https://myhost.com/index.php/system/payment/stripe/webhook

**Comment by cococonuts** [comment]
@chriskapp 
vendor/fusio/adapter-stripe/src/Provider/Stripe.php : portal(...)
$externalId = $user->getExternalId();

externalId is empty. Where can I set the externalId for user?
<img width="795" alt="image" src="https://github.com/apioo/fusio/assets/10249233/c28b515a-5d53-47c0-818e-c354061b82db">

**Comment by cococonuts** [comment]
@chriskapp I solved the billing portal problem.
I didn't find an entry-point for user cancel the subscription. How can an user cancel the subscription?

**Comment by cococonuts** [comment]
@chriskapp 
Another question:
If user buy more than one subscriptions, will the points/credit added up?

**Comment by cococonuts** [comment]
@chriskapp 
I could purchase subscription and visit billing portal with one account
But I didn't find "Cancel subscription" function on the developer App.
Do I need to implement it by myself? Could you share some details?
